src/requesthandler/RequestHandler_Config.cpp

```diff
@@ -289,9 +289,7 @@ RequestResult RequestHandler::CreateProfile(const Request &request)
 	if (std::find(profiles.begin(), profiles.end(), profileName) != profiles.end())
 		return RequestResult::Error(RequestStatus::ResourceAlreadyExists);
 
-	QMainWindow *mainWindow = static_cast<QMainWindow *>(obs_frontend_get_main_window());
-	QMetaObject::invokeMethod(mainWindow, "NewProfile", Qt::BlockingQueuedConnection,
-				  Q_ARG(QString, QString::fromStdString(profileName)));
+	obs_frontend_create_profile(QString::fromStdString(profileName));
```

Comment:
Last I checked, `obs_frontend_create_profile` takes a `const char*`, so this should probably say `profileName.c_str()` instead (the signature is `void obs_frontend_create_profile(const char *name)`).

Granted, _inside_ the API it will create a new `QString` with that, but those are internals that the caller should not care about. There is no need to use _any_ Qt types here, as from the POV of the plugin Qt might as well not exist: There is a frontend, it has an API, and you don't know whether it's Qt, GTK, SwiftUI or WinUI.